internal/controller/bucket/bucket.go

```diff
@@ -214,76 +214,81 @@ func (c *external) Create(ctx context.Context, mg resource.Managed) (managed.Ext
 		return managed.ExternalCreation{}, errors.New(errNotBucket)
 	}
 
+	backends := newBackendStatuses()
+
 	bucket.Status.SetConditions(xpv1.Creating())
+	defer setBucketStatus(bucket, backends.getBackendStatuses())
+
 	// Where a bucket has a ProviderConfigReference Name, we can infer that this bucket is to be
 	// created only on this S3 Backend. An empty config reference name will be automatically set
 	// to "default".
 	if bucket.GetProviderConfigReference() != nil && bucket.GetProviderConfigReference().Name != defaultPC {
-		return c.create(ctx, bucket)
+		return c.create(ctx, bucket, backends)
 	}
 
 	// No ProviderConfigReference Name specified for bucket, we can infer that this bucket is to
 	// be created on all S3 Backends.
-	return c.createAll(ctx, bucket)
+	return c.createAll(ctx, bucket, backends)
 }
 
-func (c *external) create(ctx context.Context, bucket *v1alpha1.Bucket) (managed.ExternalCreation, error) {
+func (c *external) create(ctx context.Context, bucket *v1alpha1.Bucket, backends *backendStatuses) (managed.ExternalCreation, error) {
 	s3Backend, err := c.getStoredBackend(bucket.GetProviderConfigReference().Name)
 	if err != nil {
 		return managed.ExternalCreation{}, err
 	}
 
+	backends.setBackendStatus(bucket.GetProviderConfigReference().Name, v1alpha1.BackendNotReadyStatus)
+
 	c.log.Info("Creating bucket on single s3 backend", "bucket name", bucket.Name, "backend name", bucket.GetProviderConfigReference().Name)
 	_, err = s3Backend.CreateBucket(ctx, s3internal.BucketToCreateBucketInput(bucket))
 	if resource.Ignore(isAlreadyExists, err) != nil {
 		return managed.ExternalCreation{}, errors.Wrap(err, errCreateBucket)
 	}
 
+	backends.setBackendStatus(bucket.GetProviderConfigReference().Name, v1alpha1.BackendReadyStatus)
+
 	return managed.ExternalCreation{}, nil
 }
 
-func (c *external) createAll(ctx context.Context, bucket *v1alpha1.Bucket) (managed.ExternalCreation, error) {
+func (c *external) createAll(ctx context.Context, bucket *v1alpha1.Bucket, backends *backendStatuses) (managed.ExternalCreation, error) {
 	if !c.backendStore.BackendsAreStored() {
 		return managed.ExternalCreation{}, errors.New(errNoS3BackendsStored)
 	}
 
 	c.log.Info("Creating bucket on all available s3 backends", "bucket name", bucket.Name)
 
-	bucketCreatedErr := make(chan error)
+	g := new(errgroup.Group)
 
 	// Create the bucket on each backend in a separate go routine
 	allBackends := c.backendStore.GetAllBackends()
-	for _, client := range allBackends {
+	for beName, client := range allBackends {
+		bn := beName
 		cl := client
-		go func(bucket *v1alpha1.Bucket) {
-			var err error
+		bucket := bucket
+
+		g.Go(func() (err error) {
+			backends.setBackendStatus(bn, v1alpha1.BackendNotReadyStatus)
```

Comment:
Should we set a lower timeout for the context of each of this goroutines, it might allow us to have a better cleanup than timing out at the same time as the parent context. 

Thoughts?

Reply:
Currently we trigger all creation calls without any control. I don't see the better cleanup what you mentioned. Could you please clarify what do you mean? Anyway it is a two sharped sword. If we decrease the timeout we should do try on more backends in the same time period. On the other hand we have a chance we cancel every request before they were able to finish (on a case of high network load for example). The question is what we more afraid of:
- Thread scaling issues (decreased timeout should handle more gorutines)
- Response latency (increased timeout should give more time to backends to answer)

An other but related question is do we want to limit the number of concurrent creations?